client/editor/map_editor.cpp

```diff
@@ -184,9 +187,21 @@ void map_editor::select_tool_tile()
     ett_wdg_active = false;
   } else {
     ui.tbut_tool_tile->setToolTip(_("Close Tile Tool"));
-    ui.stackedWidget_tools->addWidget(ett_wdg);
+    ett_wdg = new editor_tool_tile(0);
+    ui.vlayout_tools->addWidget(ett_wdg);
     ett_wdg->show();
     ett_wdg->update();
     ett_wdg_active = true;
   }
 }
+
+/**
+ * \brief Capture tile selected and pass tile structure to the editor tool
+ * tile widget
+ */
+void map_editor::tile_selected(struct tile *ptile)
+{
+  // Take the tile handed to us from mapctrl.cpp and pass it to the tile tool
+  // widget, which will do the real work.
+  ett_wdg->update_ett(ptile);
```

Comment:
How can we be sure that `ett_wdg` is active?

Reply:
because of this in mapctrl - `if (editor_is_active() && check_tile_tool())`

Reply:
added another if()